lib/pf/Authentication/Source/WindowsLiveSource.pm

```diff
@@ -26,44 +28,17 @@ has 'redirect_url' => (isa => 'Str', is => 'rw', required => 1, default => 'http
 has 'domains' => (isa => 'Str', is => 'rw', required => 1, default => 'login.live.com,auth.gfx.ms,account.live.com');
 has 'create_local_account' => (isa => 'Str', is => 'rw', default => 'no');
 
-=head2 available_actions
+=head2 lookup_from_provider_info
 
-For an oauth2 source, we limit the available actions to B<set role>, B<set access duration>, and B<set unreg date>.
+Lookup the person information from the authentication hash received during the OAuth process
 
 =cut
 
-sub available_actions {
-    return [
-            $Actions::SET_ROLE,
-            $Actions::SET_ACCESS_DURATION,
-            $Actions::SET_UNREG_DATE,
-           ];
-}
-
-=head2 available_attributes
-
-=cut
-
-sub available_attributes {
-    my $self = shift;
-    return([@{$self->SUPER::available_attributes}, {value => 'username', type => $Conditions::SUBSTRING }]);
-}
-
-=head2 match_in_subclass
-
-=cut
+sub lookup_from_provider_info {
+    my ( $self, $pid, $info ) = @_;
+    my $logger = get_logger();
```

Comment:
I'll remove those although I find it useful sometimes to have this declared everywhere so if I need to add a logging statement quickly, I don't have to do the import + declaration.